app/api/schema/discount_codes.py

```diff
@@ -90,6 +92,41 @@ def validate_quantity(self, data, original_data):
                 raise UnprocessableEntity({'pointer': '/data/attributes/tickets-number'},
                                           "tickets-number should be greater than max-quantity")
 
+    @validates_schema(pass_original=True)
+    def validate_value(self, data, original_data):
+        if 'id' in original_data['data']:
+            try:
+                discount_code = DiscountCode.query.filter_by(id=original_data['data']['id']).one()
+            except NoResultFound:
+                raise ObjectNotFound({'parameter': '{code}'}, "DiscountCode: not found")
+
+            if 'type' not in data:
+                data['type'] = discount_code.type
+
+            if 'value' not in data:
+                data['value'] = discount_code.value
+
+        if data['type'] == "percent":
+            if data['value'] < 0 or data['value'] > 100:
+                raise UnprocessableEntity(
+                    {'pointer': '/data/attributes/value'},
+                    "discount percent must be within range of 0 and 100"
+                )
+
+        if data['type'] == "amount":
+            event_object = Event.query.filter_by(id=data['event']).one()
+            for ticket in event_object.tickets:
```

Comment:
Does this mean that I cannot have any discount code if I have at least one free ticket for my event?

Reply:
@iamareebjamal currently discount codes for events is not implemented. The code is commented out in `basic-details-step`.
The discount code is implemented only on tickets for now.

Reply:
But then why are you checking for all tickets in an event?

Reply:
@iamareebjamal actually server code and tests were written so I just added to them. These are to be implemented later

Reply:
I think these implementation can be worked upon later when we start finalizing it on frontend. Till then, it can be left this way as it'll not affect by anyway @iamareebjamal

Reply:
If I merge this, nobody with an event having a free ticket will be able to create a discount code, so it is not just let's merge this and figure out later

Reply:
@iamareebjamal there is currently no option to create a discount code related to an event. 
It's just present on ther server side so no one can create any discount code related to any event, rather discount code can only be created only be associated to tickets for now.
Maybe @CosmicCoder96 can give more insight to this

Reply:
@CosmicCoder96 can you please verify that we aren't allowing creation of discount codes for events for now?